src/app/AppLayout/AuthModal.tsx

```diff
@@ -50,25 +52,35 @@ export interface AuthModalProps {
 export const AuthModal: React.FunctionComponent<AuthModalProps> = (props) => {
   const context = React.useContext(ServiceContext);
 
-  const handleDismiss = () => {
-    props.onDismiss();
-  };
-
-  const onSave = (username: string, password: string) => {
-    context.api.postTargetCredentials(username, password).pipe(first()).subscribe();
-    props.onSave();
-  };
+  const onSave = React.useCallback((username: string, password: string): Promise<void> => {
+    return new Promise((resolve) => {
```

Comment:
Should we use Observable pattern here too instead of Promise?

Reply:
Also do we need to `useSubscriptions`  for any api call on modals? Same for `src/app/SecurityPanel/Credentials/CreateJmxCredentialModal.tsx` too?

https://github.com/cryostatio/cryostat-web/blob/837a7208534191befc1b21413d333b7f0d1f1de5/src/app/utils/useSubscriptions.ts#L41

Reply:
> Should we use Observable pattern here too instead of Promise? Or is there a reason for using Promise tho?

Promise just seemed a little simpler, since this behaviour only needs to specify a callback that occurs once, not a potentially recurring observable stream. I could change it over to Observable if you prefer.

> Also do we need to useSubscriptions for any api call on modals? Same for src/app/SecurityPanel/Credentials/CreateJmxCredentialModal.tsx too?

Good catch. I'll fix that.

Reply:
Ohh I think it makes sense to use Promise for simpler use :D 

Also do you the two calls to `context.target.target` and `context.api.postCredentials` can be piped using `mergeMap` like here? That way we only 1 subscription added to `addSubcription`. Just seeing which style we prefer^^

https://github.com/cryostatio/cryostat-web/blob/837a7208534191befc1b21413d333b7f0d1f1de5/src/app/Rules/RulesUploadModal.tsx#L108

Reply:
Sure, good idea.